src/DOMComponentTree.js

```diff
@@ -37,3 +39,16 @@ export function getInternalHandleFromInstance(
   }
   return null;
 }
+
+export function collectAncestors(inst: PublicInstance): Array<PublicInstance> {
+  let handle = ComponentInstanceMap.get(inst);
+  const path: Array<PublicInstance> = [];
+  while (handle) {
+    path.push(inst);
+    do {
+      handle = handle.return;
+    } while (handle && handle.tag !== HostComponent);
+    inst = handle && handle.stateNode;
```

Comment:
Note only `HostComponent` and `HostText` are guaranteed to have an Element/Text instance there. The rest can be arbitrary objects.